dfc/callstats.go

```diff
@@ -92,16 +97,49 @@ func (c *CallStatsServer) worker() {
 			continue
 		}
 
+		var errCnt int64
+		if ci.failed {
+			errCnt = 1
+		}
+
 		s, ok := c.stats[ci.url]
 		if ok {
 			s.cnt++
-			s.total += ci.latency
+			s.totalLatency += ci.latency
+			s.errCnt += errCnt
 
 			if float64(ci.latency) > float64(c.factor)*float64(s.avg()) {
 				glog.Warningf("call %v latency %v too high, avg = %v", ci.url, ci.latency, s.avg())
 			}
 		} else {
-			c.stats[ci.url] = &latency{1, ci.latency}
+			c.stats[ci.url] = &latency{1, ci.latency, errCnt}
+		}
+
+		// from http://10.112.76.11:8080/v1/cluster/keepalive to 10_112_76_11:8080/v1/cluster/keepalive
+		// ideally 'url' is passed in as url.URL, which is not the case right now because
```

Comment:
You can also use https://golang.org/pkg/net/url/#ParseRequestURI if you want a `url.URL`

Reply:
understand, my point was the whole system should have done using url.URL when possible, not translate back and forth.